### ISSUE TITLE
Segfault processing libevent data

### ISSUE THREAD

**Issue by ghost** [DataSoft/Honeyd#19]
This was caught while running ostest:

```
Program received signal SIGSEGV, Segmentation fault.
tcp_send (con=0x811fd80, flags=4 '\004', payload=0x0, len=0) at honeyd.c:1451
1451        if(tmpl->person != NULL)
(gdb) backtrace 
#0  tcp_send (con=0x811fd80, flags=4 '\004', payload=0x0, len=0)
    at honeyd.c:1451
#1  0x08057ab1 in tcp_recv_cb (tmpl=0x0, pkt=0xb7b72054 "E\020", pktlen=52)
    at honeyd.c:2332
#2  0x08058c6b in honeyd_dispatch (tmpl=0x0, ip=0xb7b72054, iplen=52)
    at honeyd.c:2786
#3  0x080591f0 in honeyd_delay_cb (fd=-1, which=1, arg=0xbfffe124)
    at honeyd.c:768
#4  0x080534aa in honeyd_delay_packet (tmpl=0x0, ip=0xb7b72054, iplen=52, 
    src=0x0, dst=0x0, ms=0, flags=0, spoof=...) at honeyd.c:835
#5  0x08059540 in honeyd_input (inter=0x83d4978, ip=0xb7b72054, 
    iplen=<optimized out>) at honeyd.c:3015
#6  0x08059b38 in honeyd_recv_cb (ag=0x83d4978 "", pkthdr=0xbfffe374, 
    pkt=0xb7b72046 "") at honeyd.c:3170
#7  0xb7f48e64 in ?? () from /usr/lib/i386-linux-gnu/libpcap.so.0.8
#8  0xb7f4b668 in pcap_dispatch () from /usr/lib/i386-linux-gnu/libpcap.so.0.8
#9  0x08071ae4 in interface_recv (fd=14, type=2, arg=0x83d4978)
    at interface.c:516
#10 0xb7f85ce9 in event_base_loop () from /usr/lib/libevent-2.0.so.5
#11 0xb7f86a37 in event_loop () from /usr/lib/libevent-2.0.so.5
#12 0xb7f86a5b in event_dispatch () from /usr/lib/libevent-2.0.so.5
#13 0x08051d15 in main (argc=0, argv=0xbffff77c) at honeyd.c:3660
```

**Comment by DataSoft** [comment]
Found while using the "3Com 8810 switch" Fingerprint. In case that ends up mattering.

**Comment by DataSoft** [comment]
There is a template pointer variable defined as:

```
struct template *tmpl = con->tmpl;
```

The crash then takes place later down at the line:

```
if(tmpl->person != NULL)
```

It is likely that tmpl is NULL, here. As the appropriate check for this condition does not occur until after:

```
if (tmpl != NULL)
    spoof = tmpl->spoof;
else
    spoof = no_spoof;
```

This condition was not completely reliably reproducible, so I will try to make sure that this was the cause before closing out the Issue.

**Comment by DataSoft** [comment]
Fixed in 132e01bc14f6b2fe56c2f5ae8269f6624589fefc
